app/src/app/app.component.spec.ts

```diff
@@ -1,32 +1,40 @@
-import { TestBed, async } from '@angular/core/testing';
+import { TestBed, async, ComponentFixture } from '@angular/core/testing';
 
+import { EnplugService } from './enplug.service';
 import { AppComponent } from './app.component';
 
+let enplugServiceMock: any;
+let fixture: ComponentFixture<AppComponent>;
+
 describe('AppComponent', () => {
-  beforeEach(async(() => {
+  beforeEach(() => {
+    enplugServiceMock = {
+      appStatus: jasmine.createSpyObj('appStatus', ['start']),
+      on: jasmine.createSpy('on')
+    };
+
     TestBed.configureTestingModule({
-      declarations: [
-        AppComponent
-      ],
+      providers: [{ provide: EnplugService, useValue: enplugServiceMock }],
+      declarations: [AppComponent],
     }).compileComponents();
-  }));
+  });
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(AppComponent);
+  });
 
   it('should create the app', async(() => {
-    const fixture = TestBed.createComponent(AppComponent);
     const app = fixture.debugElement.componentInstance;
-    expect(app).toBeTruthy();
+    expect(app).toBeDefined();
```

Comment:
If `componentInstance` is null then this test will erroneously pass.

Reply:
At what circumstance this can be null?